pkg/web/oauthclient/token.go

```diff
@@ -75,17 +76,22 @@ func (oc *OAuthClient) freshToken(c echo.Context) (*oauth2.Token, error) {
 // HandleToken is a handler that returns a valid OAuth token.
 // It reads the token from the authorization cookie and refreshes it if needed.
 // If the cookie is not there, it returns a 401 Unauthorized error.
-func (oc *OAuthClient) HandleToken(c echo.Context) error {
-	token, err := oc.freshToken(c)
+func (oc *OAuthClient) HandleToken(w http.ResponseWriter, r *http.Request) {
+	token, err := oc.freshToken(w, r)
 	if err != nil {
-		return err
+		webhandlers.Error(w, r, err)
+		return
 	}
 
-	return c.JSON(http.StatusOK, struct {
+	w.Header().Set("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode(struct {
```

Comment:
Is it not necessary to do `w.WriteHeader(http.StatusOK)` here?

Reply:
`Write` automatically does `StatusOK` if no `WriteHeader` has been done before. And encode calls `Write`.